decode_response.go

```diff
@@ -31,9 +32,27 @@ func (sp *SAMLServiceProvider) ValidateEncodedResponse(encodedResponse string) (
 		return nil, err
 	}
 
-	response, err := sp.validationContext().Validate(doc.Root())
-	if err != nil && !sp.SkipSignatureValidation || response == nil {
-		return nil, err
+	response := doc.Root()
+	if !sp.SkipSignatureValidation {
```

Comment:
`SkipSignatureValidation` now completely bypasses `dsig.Validate`, so `ValidateEncodedResponse` wouldn't fail due to signature issues alone.